Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -592,48 +566,73 @@ private void loadDesktopBackgroundSettings()
 
             txtWindowsBackgroundPath.Text = regWallpaper;
 
-            foreach (var item in Enum.GetValues(typeof(Desktop.CairoWallpaperStyle)).Cast<Desktop.CairoWallpaperStyle>())
+            foreach (var windowsBackgroundStyleItem in Enum.GetValues(typeof(Desktop.CairoWallpaperStyle)).Cast<Desktop.CairoWallpaperStyle>())
             {
-                cboWindowsBackgroundStyle.Items.Add(item);
+                cboWindowsBackgroundStyle.Items.Add(windowsBackgroundStyleItem);
             }
 
             cboWindowsBackgroundStyle.SelectedItem = style;
 
             #endregion
-            #region  cairoImageWallpaper
-            txtCairoBackgroundPath.Text = Settings.Instance.CairoBackgroundImagePath;
 
-            foreach (var item in Enum.GetValues(typeof(Desktop.CairoWallpaperStyle)).Cast<Desktop.CairoWallpaperStyle>())
+            if (Shell.IsCairoUserShell)
```

Comment:
I think this should check for Startup.IsCairoUserShell so that the additional options don't appear for people who have set Cairo as shell but who have not yet logged out and back in. If they were to choose one of those options in this state, it wouldn't work.

Reply:
This is the last thing before merging I promise!

Reply:
Why are there two methods?

Reply:
I think I put an annotation on there but honestly we should think of a better name for one of them. `Shell.IsCairoUserShell` checks the configuration (what is in the registry) but not which shell is actually running right now. `Startup.IsCairoUserShell` checks that Cairo is both set as shell and currently running as the shell so that we can use the correct behavior.

Having `Shell.IsCairoUserShell` allows us to show the correct shell configuration state in the settings window if the user changes it but opts to not log off.

Reply:
Ya, i was just reviewing. Maybe a more complex type that will elaborate on the configuration and state. Like who is configured as shell and who is running as shell.
Also, there is a flaw in the code. We should check the status of Shell_TrayWnd on each call. I encounter this scenario quite a lot: Explorer is terminated while Cairo was running on top of it.

Reply:
It's mostly only used at startup except when changing settings. I think it should still be cached for performance reasons, but we should see if we can identify a notification sent for when Explorer closes.

Reply:
I did a thing

Reply:
The Shell_TrayWnd check needs to be cached. Cairo creates its own so this is always going to be true after we start, which is not what we want.

Reply:
Ok. Ill put it back